src/devices/Ssd1331/Ssd1331.cs

```diff
@@ -0,0 +1,1366 @@
+        public void Contrast(int contrast)
+        {
+            int v = contrast * 20;
+
+            if (v > 180)
+            {
+                v = 180;
+            }
+
+            SendCommand(Ssd1331SetContrastA, (byte)v);
+            SendCommand(Ssd1331SetContrastB, (byte)v);
+            SendCommand(Ssd1331SetContrastC, (byte)v);
+        }
+
+        /// <summary>
+        /// Set up window to be acted on
+        /// </summary>
+        /// <param name="x">Top-left column</param>
+        /// <param name="y">Top-left row</param>
+        /// <param name="w">Bottom-right column</param>
+        /// <param name="h">Bottom-right row</param>
+        public void Window(int x, int y, int w, int h)
+        {
+            _x = (byte)x;
+            _y = (byte)y;
+
+            _x1 = (byte)x;
+            _x2 = (byte)(x + w - 1);
+            _y1 = (byte)y;
+            _y2 = (byte)(y + h - 1);
+            SetColumnAddress();
+            SetRowAddress();
+            SetColumnAddress(_x1, _x2);
+            SetRowAddress(_y1, _y2);
+        }
+
+        /// <summary>
+        /// Display a 16bit bitmap
+        /// </summary>
+        /// <param name="x">Starting column</param>
+        /// <param name="y">Starting row</param>
+        /// <param name="Name_BMP">Filename of bitmap</param>
+        /// <returns></returns>
+        public int Bitmap16FS(int x, int y, string Name_BMP)
+        {
+            int offsetPixelWidth = 18;
+            int offsetPixelHeight = 22;
+            int offsetPixData = 10;
+            int offsetBPP = 28;
+            byte[] bmpHeader = new byte[54];
+            ushort bppT;
+            int pixelWidth;
+            int pixelHeight;
+            int startData;
+            int i;
+            int off;
+            int padd;
+            int j;
+            byte[] line;
+
+            FileStream image = new FileStream(Name_BMP, FileMode.Open);
+            image.Read(bmpHeader, 0, 54);
+            if (bmpHeader[0] != 0x42 || bmpHeader[1] != 0x4D)
+            {
+                image.Close();
+                return -1;
+            }
+
+            bppT = (ushort)(bmpHeader[offsetBPP] + (bmpHeader[offsetBPP + 1] << 8));
+            if (bppT != 0x0010)
+            {
+                image.Close();
+                Console.Error.WriteLine("Error, not 16 bit BMP");
+                return -2;
+            }
+
+            pixelHeight = bmpHeader[offsetPixelHeight] + (bmpHeader[offsetPixelHeight + 1] << 8) + (bmpHeader[offsetPixelHeight + 2] << 16) + (bmpHeader[offsetPixelHeight + 3] << 24);
+            pixelWidth = bmpHeader[offsetPixelWidth] + (bmpHeader[offsetPixelWidth + 1] << 8) + (bmpHeader[offsetPixelWidth + 2] << 16) + (bmpHeader[offsetPixelWidth + 3] << 24);
+            if (pixelHeight > Height + 1 + y || pixelWidth > Width + 1 + x)
+            {
+                image.Close();
+                return -3;                                        // to big

```

Comment:
can you place the comment on the line before? We tend to do this always to have more clarity in the code. And here, it's anyway far from the code :-)